client/src/components/Workflow/Import/TrsImport.vue

```diff
@@ -0,0 +1,161 @@
+import { getGalaxyInstance } from "app";
+import { safePath } from "utils/redirect";
+import { computed, ref, watch } from "vue";
+import { redirectOnImport } from "../utils";
+import LoadingSpan from "components/LoadingSpan";
+import TrsServerSelection from "./TrsServerSelection";
+import DebouncedInput from "components/DebouncedInput";
+
+const props = defineProps({
+    queryTrsServer: {
+        type: String,
+        default: null,
+    },
+    queryTrsId: {
+        type: String,
+        default: null,
+    },
+    queryTrsVersionId: {
+        type: String,
+        default: null,
+    },
+    isRun: {
+        type: Boolean,
+        default: false,
+    },
+});
+
+const trsTool = ref(null);
+const loading = ref(false);
+const importing = ref(false);
+const trsSelection = ref(null);
+const errorMessage = ref(null);
+const toolId = ref(props.queryTrsId);
+const isAnonymous = getGalaxyInstance().user?.isAnonymous();
+const isAutoImport = ref(props.queryTrsVersionId && props.queryTrsServer && props.queryTrsId);
+
+const toolIdTrimmed = computed(() => {
+    return toolId.value?.trim() || null;
+});
+const hasErrorMessage = computed(() => {
+    return errorMessage.value != null;
+});
+
+watch(toolIdTrimmed, () => {
+    onToolId();
+});
+
+const services = new Services();
+
+const onToolId = () => {
+    trsTool.value = null;
+    errorMessage.value = null;
+    if (!trsSelection.value || !toolIdTrimmed.value) {
+        return;
+    }
+    loading.value = true;
+    services
+        .getTrsTool(trsSelection.value.id, toolIdTrimmed.value)
+        .then((tool) => {
+            trsTool.value = tool;
+            if (isAutoImport.value) {
+                let versionField = "name";
+                const version = trsTool.value.versions.find((version) => {
+                    if (version.name === props.queryTrsVersionId) {
+                        return true;
+                    } else if (version.id === props.queryTrsVersionId) {
+                        versionField = "id";
+                        return true;
+                    }
+                });
+                if (version) {
+                    importVersion(trsSelection.value.id, trsTool.value.id, version[versionField], props.isRun);
+                } else {
+                    Toast.warning(`Specified version: ${props.queryTrsVersionId} doesn't exist`);
+                    isAutoImport.value = false;
+                }
+            }
+        })
+        .catch((e) => {
+            trsTool.value = null;
+            errorMessage.value = e;
+        })
+        .finally(() => {
+            loading.value = false;
+        });
+};
+
+const onTrsSelection = (selection) => {
+    trsSelection.value = selection;
+    if (toolIdTrimmed.value) {
+        onToolId();
+    }
+};
+
+const onTrsSelectionError = (message) => {
+    errorMessage.value = message;
+};
+
+const importVersion = (trsId, toolIdToImport, version = null, isRunFormRedirect = false) => {
```

Comment:
Looks like this code from the mixin is now duplicated; should be refactored out and shared

Reply:
(resolving, I plan to address this in followup refactoring here)